pkgs/by-name/co/code-cursor/package.nix

```diff
@@ -5,68 +5,114 @@
 
-  nativeBuildInputs = [ makeWrapper ];
+        substituteInPlace $out/share/applications/cursor.desktop --replace-fail "AppRun" "cursor"
 
-  installPhase = ''
-    runHook preInstall
+        wrapProgram $out/bin/cursor \
+          --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations --enable-wayland-ime=true}} --no-update"
 
-    mkdir -p $out/
-    cp -r bin $out/bin
+        runHook postInstall
+      '';
 
-    mkdir -p $out/share/cursor
-    cp -a ${appimageContents}/locales $out/share/cursor
-    cp -a ${appimageContents}/resources $out/share/cursor
-    cp -a ${appimageContents}/usr/share/icons $out/share/
-    install -Dm 644 ${appimageContents}/cursor.desktop -t $out/share/applications/
+  appimageContents = appimageTools.extractType2 { inherit version pname src; };
+in
+stdenvNoCC.mkDerivation {
+  inherit pname version src;
 
-    substituteInPlace $out/share/applications/cursor.desktop --replace-fail "AppRun" "cursor"
+  nativeBuildInputs = [ makeWrapper ] ++ lib.optional stdenvNoCC.isDarwin [ undmg ];
 
-    wrapProgram $out/bin/cursor \
-      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations --enable-wayland-ime=true}} --no-update"
+  sourceRoot = ".";
 
-    runHook postInstall
-  '';
+  inherit installPhase;
 
   passthru.updateScript = writeScript "update.sh" ''
     #!/usr/bin/env nix-shell
     #!nix-shell -i bash -p curl yq coreutils gnused common-updater-scripts
     set -eu -o pipefail
+
+    # Check both Linux and Mac versions to ensure they match
     latestLinux="$(curl -s https://download.todesktop.com/${appKey}/latest-linux.yml)"
-    version="$(echo "$latestLinux" | yq -r .version)"
-    filename="$(echo "$latestLinux" | yq -r '.files[] | .url | select(. | endswith(".AppImage"))')"
-    url="https://download.todesktop.com/${appKey}/$filename"
+    latestMac="$(curl -s https://download.todesktop.com/${appKey}/latest-mac.yml)"
+
+    linuxVersion="$(echo "$latestLinux" | yq -r .version)"
+    macVersion="$(echo "$latestMac" | yq -r .version)"
+
+    if [ "$linuxVersion" != "$macVersion" ]; then
+      echo "Warning: Linux version ($linuxVersion) and Mac version ($macVersion) differ"
+      exit 1
+    fi
+
+    version="$linuxVersion"
     currentVersion=$(nix-instantiate --eval -E "with import ./. {}; code-cursor.version or (lib.getVersion code-cursor)" | tr -d '"')
 
     if [[ "$version" != "$currentVersion" ]]; then
-      hash=$(nix-hash --to-sri --type sha256 "$(nix-prefetch-url "$url")")
-      update-source-version code-cursor "$version" "$hash" "$url" --source-key=src.src
+      echo "Updating code-cursor from $currentVersion to $version"
+      update-source-version code-cursor "$version"
+    else
+      echo "code-cursor is already up to date at version $version"
     fi
   '';
 
-  meta = {
+  meta = with lib; {
```

Comment:
`with lib` is deprecated, please remove and refer to lib inline (as per the original file).

Background: https://discourse.nixos.org/t/why-are-there-so-many-ways-to-reference-lib-and-do-they-matter/2920